server/pages/index.js

```diff
@@ -379,95 +365,8 @@ module.exports = function() {
 		}
 	} );
 
-	if ( config.isEnabled( 'manage/themes/details' ) ) {
-		app.get( '/theme/:theme_slug/:section?', function( req, res ) {
-			function updateRenderCache( themeSlug ) {
-				wpcom.undocumented().themeDetails( themeSlug, ( error, data ) => {
-					if ( error ) {
-						debug( `Error fetching theme ${ themeSlug } details: `, error.message || error );
-						return;
-					}
-					const themeData = themeDetails.get( themeSlug );
-					if ( ! themeData || ( Date( data.date_updated ) > Date( themeData.date_updated ) ) ) {
-						debug( 'caching', themeSlug );
-						themeDetails.set( themeSlug, data );
-						// update the render cache
-						renderThemeSheet( data );
-					}
-				} );
-			}
-
-			function renderThemeSheet( theme ) {
-				const context = {};
-				const store = createReduxStore();
-				store.dispatch( {
-					type: ThemesActionTypes.RECEIVE_THEME_DETAILS,
-					themeId: theme.id,
-					themeName: theme.name,
-					themeAuthor: theme.author,
-					themeScreenshot: theme.screenshot,
-					themePrice: theme.price ? theme.price.display : undefined,
-					themeDescription: theme.description,
-					themeDescriptionLong: theme.description_long,
-					themeSupportDocumentation: theme.extended ? theme.extended.support_documentation : undefined,
-				} );
-
-				store.dispatch( setSection( 'themes', { hasSidebar: false, isFullScreen: true } ) );
-				context.initialReduxState = pick( store.getState(), 'ui', 'themes' );
-
-				const primary = <ThemeDetails id={ theme.id }><ThemeSheet /></ThemeDetails>;
-
-				const element = (
-					<ReduxProvider store={ store }>
-						<LayoutLoggedOut primary={ primary } />
-					</ReduxProvider>
-				);
-				const path = url.parse( req.url ).path;
-				const key = JSON.stringify( element ) + path + JSON.stringify( context.initialReduxState );
-				Object.assign( context, render( element, key ) );
-				return context;
-			};
-
-			const context = getDefaultContext( req );
-			if ( config.isEnabled( 'server-side-rendering' ) ) {
-				const theme = themeDetails.get( req.params.theme_slug );
-				if ( theme ) {
-					Object.assign( context, renderThemeSheet( theme ) );
-					debug( 'found theme!', theme.id );
-				}
-
-				i18n.initialize();
-				req.params.theme_slug && updateRenderCache( req.params.theme_slug ); // TODO(ehg): We don't want to hit the endpoint for every req. Debounce based on theme arg?
-			}
-
-			res.render( 'index.jade', context );
-		} );
-	}
-
-	app.get( '/design(/type/:themeTier)?', function( req, res ) {
-		if ( req.cookies.wordpress_logged_in || ! config.isEnabled( 'manage/themes/logged-out' ) ) {
-			// the user is probably logged in
-			renderLoggedInRoute( req, res );
-		} else {
-			const context = getDefaultContext( req );
-			const tier = includes( [ 'all', 'free', 'premium' ], req.params.themeTier )
-				? req.params.themeTier
-				: 'all';
```

Comment:
Not urgent, but we should still whitelist `tier` in the new implementation, especially as it's a potential vector for undesired cache manipulation.